app/src/main/java/com/duckduckgo/app/privacymonitor/ui/PrivacyDashboardViewModel.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2017 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.privacymonitor.ui
+
+import android.annotation.SuppressLint
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.content.Context
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.privacymonitor.HttpsStatus
+import com.duckduckgo.app.privacymonitor.PrivacyMonitor
+import com.duckduckgo.app.privacymonitor.store.PrivacySettingsStore
+
+@SuppressLint("StaticFieldLeak")
+class PrivacyDashboardViewModel(private val context: Context,
+                                private val settingsStore: PrivacySettingsStore) : ViewModel() {
+
+    data class ViewState(
+            val domain: String,
+            val heading: String,
+            val httpsIcon: Int,
+            val httpsText: String,
+            val networksText: String,
+            val networksIcon: Int = R.drawable.dashboard_networks_good,
+            val majorNetworksText: String,
+            val majorNetworksIcon: Int = R.drawable.dashboard_major_networks_good,
+            val toggleEnabled: Boolean,
+            val toggleText: String,
+            val toggleBackgroundColor: Int = R.color.midGreen
```

Comment:
I would suggest using the `@*Res` annotations here to give compile time checking of the resource type (e.g. `@ColorRes`  if they work (don't seem to be for me)

Reply:
Good thinking